components/ResourceTable.tsx

```
@@ -0,0 +1,851 @@
+        console.log("🔍 Original storage URL:", urlToCheck);
+
+        try {
+          // Check if the token is expired or about to expire
+          let needsRegeneration = false;
+
+          if (urlToCheck) {
+            try {
+              // Extract the JWT token from the URL
+              const tokenMatch = urlToCheck.match(/token=([^&]+)/);
+              if (tokenMatch && tokenMatch[1]) {
+                const token = tokenMatch[1];
```

Biome (1.9.4)

[error] Change to an optional chain.

Unsafe fix: Change to an optional chain.


(lint/complexity/useOptionalChain)